jovian/tests/utils/test_clone.py

```diff
@@ -74,14 +77,25 @@ def test_get_gist(mock_get, gist, called_with_url):
 
         mock_get.assert_called_with(called_with_url, headers=HEADERS)
 
+def mock_get_gist(gist, version, fresh=True, *args, **kwargs):
+        first_gist=gist
+        first_version=version
+        yield get_gist(gist, version, fresh=True)
+        assert mock_get_gist.assert_called_with(first_gist, first_version, True)
```

Comment:
can we refactor this and move the assert to the test itself.

Reply:
Can you suggest a different way of handling recursive calls.

Reply:
It seems that this test is not testing anything at the moment. You can replace this assert statement with `assert False` to check if the test breaks. 

You may want to use `assert_has_calls` if you want to check if a function was called multiple times.